### PR TITLE
Create release-checklist

### PR DESCRIPTION
**Pull request by core-man**
Addresses #313, #314.

### CODE REVIEW

File: .github/workflows/ISSUE_TEMPLATE/release-checklist.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+**Release**:
+- [ ] Set the tag version to vX.Y
+- [ ] Manually upload the PDF to the release
+
+**After release**:
+- [ ] Bump the GMT version in the CI workflow: `cache-data.yml`, `deploy.yml`, and `preview-pr.yml`
+- [ ] Bump the GMT version number in `source/_static/version_switch.js`, `source/conf.py`, and `README.md`
+- [ ] Bump the GMT version in the documentation
+  - [ ] `source/intro`: `overview.rst`, `resources.rst`, `versions.rst`
```
Comment (seisman) [inline .github/workflows/ISSUE_TEMPLATE/release-checklist.md:29]: I don't think it's necessary to change `versions.rst`:
```suggestion
  - [ ] `source/intro`: `overview.rst`, `resources.rst`
```

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+**Release**:
+- [ ] Set the tag version to vX.Y
+- [ ] Manually upload the PDF to the GitHub release page
+
+**After release**:
+- [ ] Bump the GMT version in the CI workflow: `cache-data.yml`, `deploy.yml`, and `preview-pr.yml`
+- [ ] Bump the GMT version number in `source/_static/version_switch.js`, `source/conf.py`, and `README.md`
+- [ ] Bump the GMT version in the documentation
+  - [ ] `source/intro`: `overview.rst`, `resources.rst`, `versions.rst`
```
Comment (core-man) [inline .github/ISSUE_TEMPLATE/release-checklist.md:29]:
```suggestion
  - [ ] `source/intro`: `overview.rst` and `resources.rst`
```

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  - [ ] `source/install`: `macOS.rst`, `windows.rst`, `cross-platform.rst`, `build-source.rst`
```
Comment (core-man) [inline .github/ISSUE_TEMPLATE/release-checklist.md:30]:
```suggestion
  - [ ] `source/install`: `macOS.rst`, `windows.rst`, `cross-platform.rst`, and `build-source.rst`
```

File: .github/workflows/ISSUE_TEMPLATE/release-checklist.md (hunk 1 above)
Comment (core-man) [reply .github/workflows/ISSUE_TEMPLATE/release-checklist.md]: Yes, please see revision below.

**Review by seisman — APPROVED** [review]

**Pull request merged**